content/300-guides/150-testing/150-integration-testing.mdx

```diff
@@ -0,0 +1,450 @@
+</TopBlock>
+
+#### Prerequisites
+
+This guide assumes you have [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) installed on your machine as well as [`Jest`](./unit-testing#add-jest-to-project) setup in your project.
+
+The following ecommerce schema will be used throughout the guide. This varies from the traditional `User` and `Post` models used in other parts of the docs, mainly because it is unlikely you will be running integration tests against your blog.
+
+<details><summary>Ecommerce schema</summary>
+
+```prisma file=schema.prisma
+// Can have 1 customer
+// Can have many order details
+model Order {
+  id           Int            @id @default(autoincrement())
+  createdAt    DateTime       @default(now())
+  customer     Customer       @relation(fields: [customerId], references: [id])
+  customerId   Int
+  orderDetails OrderDetails[]
+}
+
+// Can have 1 order
+// Can have 1 product
+// The quantity filed dictates how many products are in the order
+model OrderDetails {
+  id        Int     @id @default(autoincrement())
+  products  Product @relation(fields: [productId], references: [id])
+  productId Int
+  order     Order   @relation(fields: [orderId], references: [id])
+  orderId   Int
+  total     Decimal
+  quantity  Int
+}
+
+// Can have many order details
+// Can have 1 category
+model Product {
+  id           Int            @id @default(autoincrement())
+  name         String
+  description  String
+  price        Decimal
+  sku          Int
+  orderDetails OrderDetails[]
+  category     Category       @relation(fields: [categoryId], references: [id])
+  categoryId   Int
+}
+
+// Can have many products
+model Category {
+  id       Int       @id @default(autoincrement())
+  name     String
+  products Product[]
+}
+
+// Can have many orders
+model Customer {
+  id      Int     @id @default(autoincrement())
+  email   String  @unique
+  address String?
+  name    String?
+  orders  Order[]
+}
+```
+
+</details>
+
+The guide uses a singleton pattern for Prisma Client setup. Refer to the [singleton](./unit-testing#singleton) docs for a walk through of how to set that up.
+
+## Add Docker to your project
+
+With Docker and Docker compose both installed on your machine, begin by creating a `docker-compose.yml` file at your projects root. Here you will add a Postgres image and specify the environments credentials.
+
+```yml file=docker-compose.yml
+# Set the version of docker compose to use
+version: '3.9'
+
+# The containers that compose the project
+services:
+  db:
+    image: postgres:13
+    restart: always
+    container_name: integration-tests-prisma
+    ports:
+      - '5433:5432'
+    environment:
+      POSTGRES_USER: prisma
+      POSTGRES_PASSWORD: prisma
+      POSTGRES_DB: tests
+```
+
+> **Note**: The version used here is the latest at the time of writing, if you are following along be sure to use the same version for consistency.
+
+Breaking down the file:
```

Comment:
Something more formal - e.g. "The `docker-compoose.yml` file defines the following:"